From ff2ca2a9974cc6e3b51e3690294922da80d784a5 Mon Sep 17 00:00:00 2001
From: Jan Janssen <janssen@mpie.de>
Date: Thu, 2 Nov 2023 09:08:31 +0100
Subject: [PATCH] remove use_2to3

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index f575020..678b555 100644
--- a/setup.py
+++ b/setup.py
@@ -36,5 +36,4 @@
       test_suite='nose.collector',
       include_package_data=True,
       zip_safe=True,
-      use_2to3=True,
       )
